fmanimashaun/school_library
# OOP school library: add basic UI

# Implement Console App Functionality

## Description

This PR introduces several new features to our console app, bringing it in line with the project requirements and UI example video. The changes are as follows:

1. **app.rb:** This file serves as the entry point for our console app. It includes methods to:
    - List all books
    - List all people
    - Create a person (either a teacher or a student)
    - Create a book
    - Create a rental
    - List all rentals for a given person id

2. **main.rb:** This file defines the main method, which is invoked at the end of the file. It includes functionality to:
    - Present the user with a list of options to perform
    - Allow users to choose an option
    - If needed, ask for parameters for the option
    - Provide a way to quit the app

## Testing Steps

To verify these changes, follow these steps:

1. Run the console app using `ruby './main.rb'`
2. Verify that all options are presented to the user.
3. Test each option to ensure they perform as expected.
4. Check edge cases, such as invalid input or empty states.
